clang/include/clang/AST/Decl.h

```diff
@@ -3208,7 +3208,7 @@ class FieldDecl : public DeclaratorDecl, public Mergeable<FieldDecl> {
 
   /// Find the FieldDecl specified in a FAM's "counted_by" attribute. Returns
   /// \p nullptr if either the attribute or the field doesn't exist.
-  const FieldDecl *findCountedByField() const;
+  FieldDecl *findCountedByField() const;
```

Comment:
It's unfortunate that we're dropping the const correctness here; why not a `non-const` member function for an overload set?

Reply:
I really hate using `const_cast` when I don't have to. :-) And as it turned out, it wasn't needed because nothing in the method required it to be const.

Reply:
Sorry, but that makes no sense to me. I need this method to return a non-const `FieldDecl *`. Using `const_cast` everywhere just so there's a 'const' version of the method, which I don't need, seems like an anti-pattern to me.